flyteplugins/go/tasks/pluginmachinery/flytek8s/k8s_resource_adds_test.go

```diff
@@ -315,17 +318,78 @@ func TestDecorateEnvVars(t *testing.T) {
 		args                  args
 		additionEnvVar        map[string]string
 		additionEnvVarFromEnv map[string]string
+		offloadingEnabled     bool
+		offloadingEnvVar      map[string]string
 		executionEnvVar       map[string]string
 		consoleURL            string
 		want                  []v12.EnvVar
 	}{
-		{"no-additional", args{envVars: defaultEnv, id: mockTaskExecutionIdentifier{}}, emptyEnvVar, emptyEnvVar, emptyEnvVar, "", expected},
-		{"with-additional", args{envVars: defaultEnv, id: mockTaskExecutionIdentifier{}}, additionalEnv, emptyEnvVar, emptyEnvVar, "", aggregated},
-		{"from-env", args{envVars: defaultEnv, id: mockTaskExecutionIdentifier{}}, emptyEnvVar, envVarsFromEnv, emptyEnvVar, "", aggregated},
-		{"from-execution-metadata", args{envVars: defaultEnv, id: mockTaskExecutionIdentifier{}}, emptyEnvVar, emptyEnvVar, additionalEnv, "", aggregated},
+		{
+			"no-additional",
+			args{envVars: defaultEnv, id: mockTaskExecutionIdentifier{}},
+			emptyEnvVar,
+			emptyEnvVar,
+			false,
+			emptyEnvVar,
+			emptyEnvVar,
+			"",
+			expected,
+		},
+		{
+			"no-additional-offloading-enabled",
+			args{envVars: defaultEnv, id: mockTaskExecutionIdentifier{}},
+			emptyEnvVar,
+			emptyEnvVar,
+			true,
+			emptyEnvVar,
+			emptyEnvVar,
+			"",
+			expectedOffloaded,
+		},
+		{
+			"with-additional",
+			args{envVars: defaultEnv, id: mockTaskExecutionIdentifier{}},
+			additionalEnv,
+			emptyEnvVar,
+			false,
+			emptyEnvVar,
+			emptyEnvVar,
+			"",
+			aggregated,
+		},
+		{
+			"from-env",
+			args{envVars: defaultEnv, id: mockTaskExecutionIdentifier{}},
+			emptyEnvVar,
+			envVarsFromEnv,
+			false,
+			emptyEnvVar,
+			emptyEnvVar,
+			"",
+			aggregated,
+		},
+		{
+			"from-execution-metadata",
+			args{envVars: defaultEnv, id: mockTaskExecutionIdentifier{}},
+			emptyEnvVar,
+			emptyEnvVar,
+			false,
+			emptyEnvVar,
+			additionalEnv,
+			"",
+			aggregated,
+		},
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
+			assert.NoError(t, propellerCfg.SetConfig(&propellerCfg.Config{
+				LiteralOffloadingConfig: propellerCfg.LiteralOffloadingConfig{
+					Enabled:                  tt.offloadingEnabled,
+					MinSizeInMBForOffloading: 1,
+					MaxSizeInMBForOffloading: 42,
+				},
+			}))
```

Comment:
nit : I think instead of introducing SetConfig, you could do this
```
			cfg := propellerCfg.GetConfig()
			cfg.LiteralOffloadingConfig = propellerCfg.LiteralOffloadingConfig{
				Enabled:                  tt.offloadingEnabled,
				MinSizeInMBForOffloading: 1,
				MaxSizeInMBForOffloading: 42,
			}
```
As GetConfig returns a pointer.